### ISSUE TITLE
Documentation?

### ISSUE THREAD

**Issue by eltiare** [ender672/ruby-vips#5]
I realize this is a "low-level library" - but is there some place where I can go to learn how to work with it? I'm a bit confused as to what functions are available, what they do, and what the input parameters mean.

**Comment by ender672** [comment]
RDoc is here: http://rubyvips.holymonkey.com/

Example is here: http://www.vips.ecs.soton.ac.uk/index.php?title=Speed_and_Memory_Use#ruby-vips

Another example here: https://github.com/ender672/ruby-vips

**Comment by eltiare** [comment]
Sorry, don't know how I missed that. :/

**Comment by ender672** [comment]
Thanks for asking. Let me know if you come across any other questions.

**Comment by eltiare** [comment]
I'm working on integration with CarrierWave, so I'm sure I'll have plenty.

**Comment by eltiare** [comment]
Actually, I got everything working that I wanted except for one thing: how does one go about detecting the current format of a loaded file?

**Comment by eltiare** [comment]
Alright, so I've pushed my changes to carrierwave here: https://github.com/eltiare/carrierwave/commit/08bc69f379b25d413a6243e6545defe2a88b45f0 I just can't figure out to make sure that it converts to another format for testing. 

You should really make this available to install via Ruby gems. It's in much better shape than most other libraries there...

**Comment by ender672** [comment]
Hmm... I don't think there is a way right now. VIPS does make this information available when reading images, but the library never exposes that information. Would be a nice feature.

The procedure that VIPS uses to determine the format of the image is simple - IIRC, it first looks at the file extension and if that doesn't result in a match it reads the first few bytes of the file looking for a signature.

Thanks for the encouragement. I stopped just short of releasing the gem when my son was born and I started pig farming.
